video/out/opengl/egl_helpers.c

```diff
@@ -192,18 +192,22 @@ static bool create_context(struct ra_ctx *ctx, EGLDisplay display,
     }
     if (!egl_ctx) {
         // Fallback for EGL 1.4 without EGL_KHR_create_context or GLES
-        // Add the context flags only for GLES - GL has been attempted above
         EGLint attrs[] = {
-            EGL_CONTEXT_CLIENT_VERSION, 2,
-            es ? EGL_CONTEXT_FLAGS_KHR : EGL_NONE, ctx_flags,
+            EGL_CONTEXT_FLAGS_KHR, ctx_flags,
+            es ? EGL_CONTEXT_CLIENT_VERSION : EGL_NONE, 2,
```

Comment:
>Because mpv already requires EGL 1.5 for desktop targets.

No we're supposed to support 1.4 as well.

Reply:
That probably should have been `>=`. Although since somehow nobody ever complained maybe it doesn't matter anymore and everyone's drivers implement 1.5 now (???). If we actually want to drop 1.4. There's a ton of code that could be dropped. Unrelated to this PR though.

Reply:
meson checks build API version. This no necessarily means that <=1.4 drivers are not supported during runtime, if there are appropriate version checks or guards.

Reply:
> Does Android use OpenGL at all?

Some devices do, but it's not an official thing.
This isn't what tripped up the Android device however, it was the `EGL_CONTEXT_FLAGS_KHR`.

> This no necessarily means that <=1.4 drivers are not supported during runtime, if there are appropriate version checks or guards.

I don't remember seeing any mentions of EGL 1.3 in mpv's codebase but we might indeed be transparently supporting it. Someone would have to check.

Regarding dropping EGL 1.4, here's what my Android 14 device says:
<pre>
[vo/gpu-next/opengl:v] EGL_VERSION=1.4 Android META-EGL
[vo/gpu-next/opengl:v] EGL_VENDOR=Android
[vo/gpu-next/opengl:v] EGL_CLIENT_APIS=OpenGL_ES
</pre>

Reply:
https://registry.khronos.org/EGL/specs/eglspec.1.4.pdf

page 44:
> The only attribute that can be specified in attrib list is EGL_CONTEXT_CLIENT_VERSION, and this attribute may only be specified when creating a OpenGL ES context (e.g. when the current rendering API is EGL_OPENGL_ES_API).

Reply:
I think you misunderstand.
It's not supposed to fall back to GLES in this function. We need a different EGLConfig and other stuff for that to work.

The code you mean is here:
https://github.com/mpv-player/mpv/blob/4ba9c8144963bfea8f019c467334d493cddb3fb7/video/out/opengl/egl_helpers.c#L246-L252